Description: Only display warnings about broken USB support when it's actually
 used (i.e. the machine has USB device filters).
Author: Felix Geyer <fgeyer@debian.org>

--- a/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
+++ b/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
@@ -55,6 +55,7 @@
 
 /* COM includes: */
 #include "CUSBController.h"
+#include "CUSBDeviceFilter.h"
 
 #if 0 /* Global USB filters are DISABLED now: */
 # define ENABLE_GLOBAL_USB
@@ -1117,7 +1118,7 @@ bool UISettingsDialogMachine::isPageAvai
             /* Get the USB controller object: */
             CUSBController controller = m_machine.GetUSBController();
             /* Show the machine error message if any: */
-            if (!m_machine.isReallyOk() && !controller.isNull() && controller.GetEnabled())
+            if (!m_machine.isReallyOk() && !controller.isNull() && controller.GetEnabled() && (controller.GetDeviceFilters().size() != 0))
                 msgCenter().cannotAccessUSB(m_machine);
             /* Check if USB is implemented: */
             if (controller.isNull() || !controller.GetProxyAvailable())
